### PR TITLE
Add Apps Table View to Homepage

### PR DESCRIPTION
**Pull request by kildre**
Created a new AppTable component. 
## Reference Issues or PRs
https://gitlab.jatic.net/jatic/team-metrostar/t-e-platform/-/issues/386

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [X] Did you test the pull request locally?
- [X] Did you add new tests?

#### Text styling

- [X] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [X] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [X] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "@emotion/styled": "^11.11.5",
     "@mui/icons-material": "^5.15.15",
     "@mui/material": "^5.15.10",
+    "@mui/x-data-grid": "^7.1.1",
```
Comment (jbouder) [inline ui/package.json:23]: This should be removed as it seems to be breaking the MUI Core Switch

File: ui/src/data/api.ts [hunk 2]
```diff
@@ -284,7 +284,7 @@ export const userState: UserState = {
   groups: ['group1', 'group2'],
   kind: 'user',
   last_activity: null,
-  name: 'test',
+  name: 'Longwaithe Dickenson',
```
Comment (jbouder) [inline ui/src/data/api.ts:287]: Who is this? ...just curious :)

File: ui/src/pages/home/apps-section/apps-section.tsx [hunk 3]
```diff
@@ -23,17 +23,37 @@ import {
...
+    currentNotification,
+  );
   const [currentSortValue] = useRecoilState<string>(defaultSortValue);
-  const [apps, setApps] = useState<JhApp[]>([]);
+
+  const toggleView = () => setIsGridViewActive((prev) => !prev); // Added toggleView function
+
+  useEffect(() => {
+    const serverStatus = apps ? apps.map((app) => app.status) : [];
+    console.log('APPSa', apps);
```
Comment (jbouder) [inline ui/src/pages/home/apps-section/apps-section.tsx:50]: Remove this

File: ui/src/pages/home/home.tsx [hunk 4]
```diff
@@ -1,10 +1,262 @@
...
 import { Item } from '../../styles/styled-item';
 import { AppsSection } from './apps-section/apps-section';
 import { ServicesSection } from './services-section/services-section';
 
 export const Home = (): React.ReactElement => {
+  const [, setNotification] = useRecoilState<string | undefined>(
+    currentNotification,
+  );
+  const [currentApp] = useRecoilState<JhApp | undefined>(defaultApp);
+  // const [updatedApps, setUpdatedApps] = useState<JhApp[]>(apps); // Added updatedApps state
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:32]: Remove this

File: ui/src/pages/home/home.tsx [hunk 5]
```diff
@@ -1,10 +1,262 @@
...
+  const [submitting, setSubmitting] = useState(false);
+  const queryClient = useQueryClient();
+  const handleStartRequst = async ({ id }: AppQueryPostProps) => {
+    try {
+      const response = await axios.post(`/server/${id}`);
+      return response;
+    } catch (error) {
+      console.error('There was an error!', error);
+      setNotification((error as Error).toString());
+      // setAppStatus('Error'); // Set status back to Ready if there's an error
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:45]: Remove this

File: ui/src/pages/home/home.tsx [hunk 6]
```diff
@@ -1,10 +1,262 @@
...
+    currentNotification,
+  );
+  const [currentApp] = useRecoilState<JhApp | undefined>(defaultApp);
+  // const [updatedApps, setUpdatedApps] = useState<JhApp[]>(apps); // Added updatedApps state
+  const [isStartOpen, setIsStartOpen] = useRecoilState(isStartOpenState);
+  const [isStopOpen, setIsStopOpen] = useRecoilState(isStopOpenState);
+  const [isDeleteOpen, setIsDeleteOpen] = useRecoilState(isDeleteOpenState);
+  const [submitting, setSubmitting] = useState(false);
+  const queryClient = useQueryClient();
+  const handleStartRequst = async ({ id }: AppQueryPostProps) => {
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:38]: Request spelled wrong, make sure to update the other instances as well.

File: ui/src/pages/home/home.tsx [hunk 7]
```diff
@@ -1,10 +1,262 @@
...
+      const response = await axios.post(`/server/${id}`);
+      return response;
+    } catch (error) {
+      console.error('There was an error!', error);
+      setNotification((error as Error).toString());
+      // setAppStatus('Error'); // Set status back to Ready if there's an error
+    }
+  };
+
+  const handleDeleteRequst = async ({ id, remove }: AppQueryDeleteProps) => {
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:49]: Request spelled wrong

File: ui/src/pages/home/home.tsx [hunk 8]
```diff
@@ -1,10 +1,262 @@
...
+      });
+      return response;
+    } catch (error) {
+      console.error('There was an error!', error);
+      setNotification((error as Error).toString());
+      // setAppStatus('Error'); // Reflect an error state
+    }
+  };
+
+  const handleStopRequst = async (id: string) => {
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:64]: Request spelled wrong

File: ui/src/pages/home/home.tsx [hunk 9]
```diff
@@ -1,10 +1,262 @@
...
+      const response = await axios.delete(`/server/${id}`, {
+        params: {
+          remove,
+        },
+      });
+      return response;
+    } catch (error) {
+      console.error('There was an error!', error);
+      setNotification((error as Error).toString());
+      // setAppStatus('Error'); // Reflect an error state
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:60]: Remove this

File: ui/src/pages/home/home.tsx [hunk 10]
```diff
@@ -1,10 +1,262 @@
...
+  };
+
+  const handleStopRequst = async (id: string) => {
+    try {
+      const response = await axios.post(`/server/${id}/stop`);
+      return response;
+    } catch (error) {
+      console.error('There was an error!', error);
+      setNotification((error as Error).toString());
+      // setAppStatus('Error'); // Reflect an error state
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:71]: Remove this

File: ui/src/pages/home/home.tsx [hunk 11]
```diff
@@ -1,10 +1,262 @@
...
+
+          // Fetch the apps again
+          const apps = await queryClient.getQueryData<JhApp[]>(['app-state']);
+
+          // Check if the deleted app still exists in the list
+          if (apps && !apps.find((app) => app.id === id)) {
+            console.log(`App with ID: ${id} has been successfully deleted.`);
+          } else {
+            console.log(`Failed to delete app with ID: ${id}.`);
+          }
```
Comment (jbouder) [inline ui/src/pages/home/home.tsx:110]: Remove these console logs

File: ui/src/store.ts [hunk 12]
```diff
@@ -24,7 +24,14 @@ const currentNotification = atom<string | undefined>({
...
 });
-
+const currentApp = atom<JhApp | undefined>({
+  key: 'currentApp',
+  default: undefined,
+});
+// const currentApp = atom<JhApp | null>({
+//   key: 'currentApp',
+//   default: null,
+// });
```
Comment (jbouder) [inline ui/src/store.ts:34]: Remove this

File: ui/src/pages/home/apps-section/app-table/app-table.test.tsx [hunk 13]
```diff
@@ -0,0 +1,455 @@
...
+        <QueryClientProvider client={queryClient}>
+          <AppTable apps={mockApps} />
+        </QueryClientProvider>
+      </RecoilRoot>,
+    );
+
+    const editButtons = Array.from(
+      within(baseElement).queryAllByTestId('EditRoundedIcon'),
+    ) as HTMLButtonElement[];
+    console.log('EDIT BUTTON LENGTH', editButtons.length);
```
Comment (jbouder) [inline ui/src/pages/home/apps-section/app-table/app-table.test.tsx:213]: Remove this

**Review by jbouder — CHANGES_REQUESTED** [review]
Mostly cleanup

File: ui/src/pages/home/apps-section/app-table/app-table.tsx [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+                        setIsDeleteOpen(true);
+                        setCurrentApp(app);
+                      }}
+                      color="inherit"
+                      size="small"
+                      className="action-button"
+                      data-testid="DeleteRoundedIcon"
+                    >
+                      <DeleteRoundedIcon />
+                    </Button>
```
Comment (jbouder) [inline ui/src/pages/home/apps-section/app-table/app-table.tsx:159]: These buttons should be visible and/or disabled as follows:

- Stop Icon: if app.status === "Running"
-- Disable button if Shared
- Start Icon: show in all other cases
-- Disable if app.status === "Pending" or app.status === "Unknown"
- Edit Icon
-- Disable if Shared
- Delete Icon
-- Disable if Shared

File: ui/src/data/api.ts (hunk 2 above)
Comment (kildre) [reply ui/src/data/api.ts:287]: Just a made up name.

**Review by jbouder — APPROVED** [review]
🚀

**Pull request merged**